### DISCUSSION TITLE
Welcome to homebridge-wol discussions!

### DISCUSSION THREAD

**Discussion by AlexGustafsson** [AlexGustafsson/homebridge-wol#132] [General]
## Hi!

We'll be using Discussions as a place to help new users, share configurations etc. and free up issues for bugs and feature tracking.

Feel free to
* Ask questions about usage etc.
* Share ideas
* Give suggestions
* Do anything else that comes to mind!

**Comment by alprio** [comment]
Hello , 
I am trying to make the plugin to work but I can't.
I have set Homebridge to run as sudo.
Plugin every time shows that Mac is offline but its not.
Freenas , latest . Ping from jail is working. Wol from jail is working.

 "name": "iMac 19.1",
            "ip": "192.168.1.10",
            "pingInterval": 2,
            "pingsToChange": 5,
            "pingTimeout": 1,
            "pingCommand": "ssh 192.168.1.10 'if [[ $(pmset -g powerstate IODisplayWrangler | tail -1 | cut -c29) -lt 4 ]]; then; exit 1; else; echo 1; fi;'",
            "pingCommandTimeout": 0,
            "mac": "b0:6e:bf:5f:f0:67",
            "broadcastAddress": "192.168.1.255",
            "startCommandTimeout": 0,
            "wakeGraceTime": 45,
            "wakeCommand": "ssh 192.168.1.10 caffeinate -u -t 300",
            "wakeCommandTimeout": 0,
            "shutdownCommand": "ssh 192.168.1.10 pmset sleepnow",
            "shutdownGraceTime": 15,
            "shutdownCommandTimeout": 0,
            "log": true,
            "logPinger": false,
            "debugLog": false,
            "returnEarly": false,
            "accessory": "NetworkDevice"

Thats my config.
Any help appreciate it.

**Reply by AlexGustafsson** [reply]
Are you sure you can SSH to the mac without a password? Does the SSH command work if you execute it in a terminal? Does the logs mention the reason why it does not work?

**Reply by alprio** [reply]
Figured it out. it works rn. 
Forgot to put name@ip and command. no sudo needed for me. Thanks
+ no need to put ping command. it works and generates states without it.

**Reply by AlexGustafsson** [reply]
Great! Feel free to return if you have more questions about the plugin.

**Comment by alprio** [comment]
Hello again ,
I uninstalled the plugin by mistake and reinstalled it. 
I used the same conf as before and settings and I get this weird error. 
sleep was working fine. passwordless login is working through freenas.(checked) Command works through freenas.(checked) when I press the plugin button I get this error.

[4/29/2021, 12:55:14 AM] [iMac - Alexandros] NetworkDevice shutdown cycle started for "iMac - Alexandros" (192.168.1.10)
[4/29/2021, 12:55:14 AM] [iMac - Alexandros] NetworkDevice "iMac - Alexandros" (192.168.1.10) went from status "Online" to "Shutting Down"
[4/29/2021, 12:55:14 AM] [iMac - Alexandros] Attempting to shut down "iMac - Alexandros" (192.168.1.10) using "ssh alprio@192.168.1.10 pmset sleepnow"
[4/29/2021, 12:55:14 AM] [iMac - Alexandros] An error occured while trying to shut down "iMac - Alexandros" (192.168.1.10): Error: Command failed: ssh alprio@192.168.1.10 pmset sleepnow
/bin/sh: ssh alprio@192.168.1.10 pmset sleepnow: not found

    at ChildProcess.exithandler (node:child_process:326:12)
    at ChildProcess.emit (node:events:369:20)
    at maybeClose (node:internal/child_process:1067:16)
    at Socket.<anonymous> (node:internal/child_process:453:11)
    at Socket.emit (node:events:369:20)
    at Pipe.<anonymous> (node:net:666:12) {
  killed: false,
  code: 127,
  signal: null,
  cmd: 'ssh alprio@192.168.1.10 pmset sleepnow',
  stdout: '',
  stderr: '/bin/sh: ssh alprio@192.168.1.10 pmset sleepnow: not found\n'
}
[4/29/2021, 12:55:14 AM] [iMac - Alexandros] Waiting for shutdown grace time (25000ms) for "iMac - Alexandros" (192.168.1.10)
[4/29/2021, 12:55:39 AM] [iMac - Alexandros] NetworkDevice "iMac - Alexandros" (192.168.1.10) went from status "Shutting Down" to "Online"

**Reply by AlexGustafsson** [reply]
`pmset sleepnow` works for me on Big Sur. Could you share your configuration?

**Reply by AlexGustafsson** [reply]
Sorry, I noticed you had already provided it. I unfortunately see no issue with your sleep command. The error you are getting, however, is due to `/bin/sh` receiving the full `ssh alprio@192.168.1.10 pmset sleepnow` as a command, which it is not. In this case `ssh` is the command and the rest are arguments. Could you check your version of node using `node -v` on the host running Homebridge? If it's a container, try to run the command in the container.

**Reply by alprio** [reply]
node version is 15.14.0

conf is :

 "name": "iMac - Alexandros",
            "ip": "192.168.1.10",
            "pingInterval": 2,
            "pingsToChange": 5,
            "pingTimeout": 1,
            "pingCommandTimeout": 0,
            "mac": "B06EBF5FF067",
            "startCommandTimeout": 0,
            "wakeGraceTime": 45,
            "wakeCommandTimeout": 0,
            "shutdownCommand": "ssh alprio@192.168.1.10 pmset sleepnow",
            "shutdownGraceTime": 25,
            "shutdownCommandTimeout": 0,
            "log": true,
            "logPinger": false,
            "debugLog": false,
            "returnEarly": false,
            "accessory": "NetworkDevice"

a video that command works inside my homebridge setup 
https://streamable.com/orh11m

**Reply by AlexGustafsson** [reply]
Yeah, this seems to be some sort of bug. But it's unclear in what part of the software. I will open up an issue to track it.

**Comment by Tapsel99** [comment]
Hi, i had problems starting Windows 10 because I didn't new what the broadcast address was and so I put in the ip address but now I found out that the normal broadcast address is 255.255.255.255, I think it it is better if you put this address in the description as an example instead of the current one.

**Reply by AlexGustafsson** [reply]
Thank you for your comment. I've clarified the use for `broadcastAddress` in the README, as well as adding a section explaining that the configurations only showcase different possibilities. I've kept the broadcast example as it showcases how one may set a non-standard broadcast address.

See this commit for changes: https://github.com/AlexGustafsson/homebridge-wol/commit/673f1820de16bd7b50e70500673e73b360b1f130
